plugins/clipboard/plugin.js

```diff
@@ -2539,6 +2583,227 @@
+
+			// If we are getting different type we need to check inside data comment if it is stored there.
+			} else {
+				var result = this._extractDataComment( this._getData( this._customDataFallbackType ) );
+				if ( result.data && result.data[ type ] ) {
+					value = result.data[ type ];
+
+				// And then fallback to regular `getData`.
+				} else {
+					value = this._getData( type );
+				}
+			}
+
+			return value !== null ? value : '';
+		},
+
+		/**
+		 * Sets given data in native `dataTransfer` object. If given MIME type is not supported it uses
+		 * {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer#_customDataFallbackType} MIME type
+		 * to save data using special comment format:
+		 *
+		 * 		<!--cke-data:{ type: value }-->
+		 *
+		 * It is important to keep in mind that `{ type: value }` object is stringified (using `JSON.stringify`)
+		 * and encoded (using `encodeURIComponent`).
+		 *
+		 * @param {String} type
+		 * @param {String} value
+		 */
+		setData: function( type, value ) {
+			// Extract custom data if used type is the same one which is used for storing it.
+			if ( type === this._customDataFallbackType ) {
+				var content = this._extractDataComment( this._getData( this._customDataFallbackType ) );
+
+				if ( content && content.data ) {
+					value = this._applyDataComment( value, content.data );
+				}
+			}
+
+			try {
+				this._nativeDataTransfer.setData( type, value );
+			} catch ( e ) {
+				if ( this._isUnsupportedMimeTypeError( e ) ) {
+					// We need to get the data first to append `type` and not to overwrite whole custom data.
+					var current = this._extractDataComment( this._getData( this._customDataFallbackType ) );
+
+					if ( !current.data ) {
+						current.data = {};
+					}
+					current.data[ type ] = value;
+
+					try {
+						this._nativeDataTransfer.setData( this._customDataFallbackType, this._applyDataComment( current.content, current.data ) );
+					} catch ( e ) {
+						// Some dev logger should be added here.
+					}
+				}
+			}
+		},
+
+		/**
+		 * Native getData wrapper.
+		 *
+		 * @private
+		 * @param {String} type
+		 * @returns {String}
+		 */
+		_getData: function( type ) {
+			try {
+				return this._nativeDataTransfer.getData( type );
+			} catch ( e ) {
+				return null;
+			}
+		},
+
+		/**
+		 * Whether provided error means that unsupported MIME type was used when calling native `dataTransfer.setData` method.
+		 *
+		 * @private
+		 * @param {Object} error
+		 * @returns {Boolean}
+		 */
+		_isUnsupportedMimeTypeError: function( error ) {
+			return error.message && error.message.search( /element not found/gi ) !== -1;
+		},
+
+		/**
+		 * Extracts `cke-data` comment from the given content.
+		 *
+		 * @private
+		 * @param {String} content
+		 * @returns {Object} Returns an object containing extracted data as `data`
+		 * and content (without `cke-data` comment) as `content`.
+		 * @returns {Object} return.data
```

Comment:
This could be a `null` - then it means that the `cke-data` comment is not present. We should mention this.